File: di-minihito-1/recursos minihitos/scripts/encriptador.cs
using System;
using System.Security.Cryptography;
using System.Text;

class Program
{
    static string EncryptMD5(string text)
    {
        using (MD5 md5 = MD5.Create())
        {
            byte[] inputBytes = Encoding.UTF8.GetBytes(text);
            byte[] hashBytes = md5.ComputeHash(inputBytes);
            return BitConverter.ToString(hashBytes).Replace("-", "").ToLower();
        }
    }

    static string EncryptSHA256(string text)
    {
        using (SHA256 sha256 = SHA256.Create())
        {
            byte[] inputBytes = Encoding.UTF8.GetBytes(text);
            byte[] hashBytes = sha256.ComputeHash(inputBytes);
            return BitConverter.ToString(hashBytes).Replace("-", "").ToLower();
        }
    }

    static string EncryptSHA1(string text)
    {
        using (SHA1 sha1 = SHA1.Create())
        {
            byte[] inputBytes = Encoding.UTF8.GetBytes(text);
            byte[] hashBytes = sha1.ComputeHash(inputBytes);
            return BitConverter.ToString(hashBytes).Replace("-", "").ToLower();
        }
    }

    static void Main(string[] args)
    {
        Console.Write(" > Introduce la cadena a encriptar usando MD5: ");
        string textoOriginal = Console.ReadLine();

        string hashMD5 = EncryptMD5(textoOriginal);
        string hashSHA256 = EncryptSHA256(textoOriginal);
        string hashSHA1 = EncryptSHA1(textoOriginal);

        Console.WriteLine($"Texto original: {textoOriginal}");
        Console.WriteLine($"Hash MD5: {hashMD5}");
        Console.WriteLine($"Hash SHA256: {hashSHA256}");
        Console.WriteLine($"Hash SHA1: {hashSHA1}");
    }
}

File: unidad-5/proyecto-gestpro/recursos minihitos/scripts/encriptador.py
import hashlib

def encrypt_md5(text):
    # Convertimos el texto a bytes
    text_bytes = text.encode('utf-8')
    # Creamos un objeto hash MD5
    md5_hash = hashlib.md5()
    # Actualizamos el objeto hash con los bytes del texto
    md5_hash.update(text_bytes)
    # Obtenemos el hash en formato hexadecimal
    return md5_hash.hexdigest()
  
def encrypt_sha256(text):
    # Convertimos el texto a bytes
    text_bytes = text.encode('utf-8')
    # Creamos un objeto hash SHA256
    sha256_hash = hashlib.sha256()
    # Actualizamos el objeto hash con los bytes del texto
    sha256_hash.update(text_bytes)
    # Obtenemos el hash en formato hexadecimal
    return sha256_hash.hexdigest()

def encrypt_sha1(text):
    # Convertimos el texto a bytes
    text_bytes = text.encode('utf-8')
    # Creamos un objeto hash SHA-1
    sha1_hash = hashlib.sha1()
    # Actualizamos el objeto hash con los bytes del texto
    sha1_hash.update(text_bytes)
    # Obtenemos el hash en formato hexadecimal
    return sha1_hash.hexdigest()

def __main():
  # Ejemplo de uso
  texto_original = str(input(" > Introduce la cadena a encriptar usando MD5: "))
  hash_md5 = encrypt_md5(texto_original)
  hash_sha256 = encrypt_sha256(texto_original)
  hash_sha1 = encrypt_sha1(texto_original)
  print(f"Texto original: {texto_original}")
  print(f"Hash MD5: {hash_md5}")
  print(f"Hash SHA256: {hash_sha256}")
  print(f"Hash SHA1: {hash_sha1}")

__main()